libs/clients/regulations-admin/src/lib/RegulationsAdminClientService.ts

```diff
@@ -1,62 +1,194 @@
+export class RegulationsAdminClientService {
   constructor(
     @Inject(RegulationsAdminClientConfig.KEY)
     private readonly config: ConfigType<typeof RegulationsAdminClientConfig>,
-  ) {
-    super()
-    this.baseURL = `${this.config.baseApiUrl}`
-    this.initialize({} as DataSourceConfig<any>)
-  }
+    private draftAuthorApi: DraftAuthorApi,
+    private draftRegulationCancelApi: DraftRegulationCancelApi,
+    private draftRegulationChangeApi: DraftRegulationChangeApi,
+    private draftRegulationsApi: DraftRegulationsApi,
+  ) {}
 
-  willSendRequest(request: RequestOptions) {
-    this.memoizedResults.clear()
-    request.headers.set('Content-Type', 'application/json')
+  draftAuthorApiWithAuth(auth: Auth) {
+    return this.draftAuthorApi.withMiddleware(new AuthMiddleware(auth))
   }
 
-  async getDraftRegulations(authorization: string, page?: number) {
-    return await this.get<TaskListType>(
-      '/draft_regulations',
-      {
-        page,
-      },
-      {
-        headers: { authorization },
-      },
+  draftRegulationCancelApiWithAuth(auth: Auth) {
+    return this.draftRegulationCancelApi.withMiddleware(
+      new AuthMiddleware(auth),
     )
   }
 
-  async getShippedRegulations(authorization: string) {
-    return await this.get<ShippedSummary[]>(
-      `/draft_regulations_shipped`,
-      {},
-      {
-        headers: { authorization },
-      },
+  draftRegulationChangeApiWithAuth(auth: Auth) {
+    return this.draftRegulationChangeApi.withMiddleware(
+      new AuthMiddleware(auth),
     )
   }
 
+  draftRegulationsApiWithAuth(auth: Auth) {
+    return this.draftRegulationsApi.withMiddleware(new AuthMiddleware(auth))
+  }
+
+  async getDraftRegulations(auth: Auth, page?: number) {
+    const allDrafts = await this.draftRegulationsApiWithAuth(
+      auth,
+    ).draftRegulationControllerGetAll({
+      page: page ?? 0,
+    })
+    return allDrafts
+  }
+
+  async getShippedRegulations(auth: Auth) {
+    return await this.draftRegulationsApiWithAuth(
+      auth,
+    ).draftRegulationControllerGetAllShipped()
+  }
+
   async getDraftRegulation(
     draftId: string,
-    authorization: string,
+    auth: Auth,
   ): Promise<RegulationDraft | null> {
-    const response = await this.get<RegulationDraft | null>(
-      `/draft_regulation/${draftId}`,
-      {},
-      {
-        headers: { authorization },
+    const draftRegulation = ((await this.draftRegulationsApiWithAuth(
+      auth,
+    ).draftRegulationControllerGetById({
+      id: draftId,
+    })) as unknown) as RegulationDraft
+
+    return draftRegulation
+  }
+
+  async getImpactsByName(
+    name: string,
+    auth: Auth,
+  ): Promise<DraftImpact[] | null> {
+    const res = ((await this.draftRegulationsApiWithAuth(
+      auth,
+    ).draftRegulationControllerGetImpactsByName({
+      name,
+    })) as unknown) as DraftImpact[]
+
+    return res
+  }
+
+  async create(auth: Auth, input: CreateDraftRegulationDto): Promise<any> {
```

Comment:
You should be able to use the return type of the `draftRegulationControllerCreate` instead of `any` to define a proper return type.